server/game/core/ongoingEffect/OngoingCardEffect.ts

```diff
@@ -1,11 +1,19 @@
-const OngoingEffect = require('./OngoingEffect.js');
-const { RelativePlayer, WildcardLocation, WildcardCardType } = require('../Constants.js');
-const EnumHelpers = require('../utils/EnumHelpers.js');
-const Contract = require('../utils/Contract.js');
-const Helpers = require('../utils/Helpers.js');
+import OngoingEffect from './OngoingEffect';
+import { RelativePlayer, WildcardLocation, WildcardCardType, CardTypeFilter, LocationFilter } from '../Constants';
+import * as EnumHelpers from '../utils/EnumHelpers';
+import * as Contract from '../utils/Contract';
+import * as Helpers from '../utils/Helpers';
+import Game from '../Game';
+import { Card } from '../card/Card';
+import { IOngoingEffectProps } from '../../Interfaces';
+import { OngoingEffectImpl } from './effectImpl/OngoingEffectImpl';
 
 class OngoingCardEffect extends OngoingEffect {
```

Comment:
One thing that might be worth looking at would be seeing if you can remove or integrate IOngoingCardEffect. There's a comment note on it but the tl;dr is in jigoku, IOngoingCardEffect is used in a lot of places in place of OngoingCardEffect to hide the details of type stuff so they wouldn't have to deal with the fact that the inner effect impl could be any type at all.

I tried removing it once but very quickly discovered it was a very deep rabbit hole, if you find yourself doing the same we can leave it for now and the current change is still a really useful incremental step forward.

Reply:
This definitely derived from trying to untangle IOngoingCardEffect, I think it and IOngoingPlayerEffect could be consolidated but it was a bit of digging, and then I reached the classes and realized they weren't TS yet. So this initial pass will just be getting the classes up to snuff, then I can work getting the interfaces next.

Reply:
Yup sgtm